### ISSUE TITLE
Setting K (node sampling depth) in SAGEConv

### ISSUE THREAD

**Issue by rmtr** [rusty1s/pytorch_geometric#1398]
## ❓ Questions & Help
Following your previous guidance, I was able to implement the SAGEConv for supervised learning with my own loss function (based on the generated embedding and groundtruth) and a custom dataset. However, I can't seem to find a way to set the depth K (i.e. number of aggregator functions) of a SAGEConv layer. Could you please guide me through this? The code can be seen below. Please let me know whether my overall code looks okay as well.

<pre>

class MyGraphSAGE(torch.nn.Module):
    def __init__(self,FeatureDim, Dataset):
        super(MyGraphSAGE,self).__init__()
        self.Dataset = Dataset
        self.conv1 = SAGEConv(FeatureDim, 32)
        self.conv2 = SAGEConv(32, 32) #i.e. I expect to get 32 dimension embedding

    def forward(self,x,EdgeIndex):
        x, EdgeIndex = x,EdgeIndex
        x = self.conv1(x, EdgeIndex)
        x = F.relu(x)
        x = F.normalize(x, p=2, dim=-1)
        x = self.conv2(x, EdgeIndex)
        x = F.relu(x)
        x = F.normalize(x, p=2, dim=-1)
        return x

# Loss function
def MyLossFunction(Embedding, GT, Adjacency):
    # Use the passed parameters to claculate the loss
    return Loss

# Training function
def train():

    model.train()
    num_epoch = 10

    Total_Loss = torch.tensor(1E-10, dtype=torch.float64)
    Loss = torch.tensor(1E-10, dtype=torch.float64, requires_grad=True).to(device)

    for epoch in range(num_epoch):
        for GraphData in TrainLoader:
            GraphData.to(device)
            optimiser.zero_grad()
            out = model(GraphData.x, GraphData.edge_index)
            Loss = MyLossFunction(out, GraphData.y, GraphData.Adjacency)
            Loss.backward()
            Total_Loss = Total_Loss + Loss
            print(Total_Loss)
            optimiser.step()

device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')

C = CreatePyGDataset.CreatePyGDataset(config) # This is a class which return two lists of torch_geometric.data.Data for training and validation
Train_Data_list, Val_Data_List = C.load_dataSet()
TrainLoader = DataLoader(Train_Data_list, batch_size=1)
ValLoader  = DataLoader(Val_Data_List, batch_size=1)

model = MyGraphSAGE(258, TrainLoader)
model = model.to(device)
optimiser = torch.optim.Adam(model.parameters(), lr=0.001, weight_decay=0.01)
train()
print("Completed")
</pre>

**Comment by rusty1s** [comment]
I am not really sure what you mean by this. If you want to apply SAGEConv with sampling, please have a look at the reddit.py example.

**Comment by rmtr** [comment]
Let me clarify. The following figure is from the GraphSAGE paper. 
![graphsage](https://user-images.githubusercontent.com/47621598/86502902-a9165a00-bdeb-11ea-9dd9-86bf4f42a22d.png)

My question is about the parameter `K` in the above figure. I can't seems to find a way to set it in the SAGEConv layer. Note that I am not referring to the sampling of the nodes (using random walks or any other method). My concern is about the size of the neighbourhood used to aggregate the node features. Hope this is clear.

**Comment by rusty1s** [comment]
So `k` corresponds to the number of hops, which is equivalent to the number of `SAGEConv` layers. In your current model, you have two `SAGEConv` layers, so `k=2`.

**Comment by rmtr** [comment]
Thank you.
My understanding of GraphSAGE seems to be wrong then :/ I will close this issue.
